services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java

```diff
@@ -135,15 +135,39 @@ public Response getMainPage() throws IOException, FreeMarkerException {
             
             boolean isTotpConfigured = false;
             boolean deleteAccountAllowed = false;
+            boolean manageAccountAllowed = false;
+            boolean manageAccountLinkAllowed = false;
+            boolean manageConsentAllowed = false;
+            boolean manageAccountBasicAuthAllowed = false;
+            boolean manageAccount2faAllowed = false;
+            boolean viewApplicationAllowed = false;
             if (user != null) {
                 isTotpConfigured = session.userCredentialManager().isConfiguredFor(realm, user, realm.getOTPPolicy().getType());
                 RoleModel deleteAccountRole = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).getRole(AccountRoles.DELETE_ACCOUNT);
                 deleteAccountAllowed = deleteAccountRole != null && user.hasRole(deleteAccountRole) && realm.getRequiredActionProviderByAlias(DeleteAccount.PROVIDER_ID).isEnabled();
+                RoleModel manageAccount = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).getRole(AccountRoles.MANAGE_ACCOUNT);
```

Comment:
By adding scope on all the account client roles to account-console client the roles are available in the token and can be read directly by the admin console rather than having to pass these as variables in the landing page for the console.

Reply:
Although I agree that this is best solution, I need Keycloak team help.
getToken() function exists in 'themes/src/main/resources/theme/keycloak.v2/account/src/app/keycloak-service/keycloak.service.ts' for getting token. How can I use this function or something similar in index.ftl in order to hide some parts of themes/src/main/resources/theme/keycloak.v2/account/resources/content.json? For tsx files, I believe change can be implemented.

Reply:
@ssilvert @jonkoops @edewit anyone that can provide some pointers here?

Reply:
@cgeorgilakis As you know, there is a "hidden" attribute on each ContentItem in content.json.

When the account console starts up, the value of that attribute is evaluated as javascript.  So you should be able to call out to a function that reads the token from there.  As long as your function returns a boolean value `true`, the ContentItem will be hidden.